src/alexapi/device_platforms/magicmirror.py

```diff
@@ -0,0 +1,172 @@
+from __future__ import print_function
+import time
+import os
+from abc import ABCMeta
+
+from baseplatform import BasePlatform
+import alexapi.bcolors as bcolors
+import BaseHTTPServer
+import threading
+import urllib2
+import urlparse
+
+"""
+Magic Mirror platform
+-----------------------------------------------------------------------------------------------------------------
+This device platform is made to integrate with the Magic Mirror project: https://github.com/MichMich/MagicMirror
+Specifically, this communicates with the MMM-AlexaPi module: https://github.com/dgonano/MMM-AlexaPi
+The two systems communicate over GET requests which,
+	- Update the Magic Mirror display with AlexaPi's status (No connection, idle, listening, processing, speaking)
+	- Allow the Magic Mirror to trigger a 'start listening' request
+"""
+
+class MagicmirrorPlatform(BasePlatform):
+	__metaclass__ = ABCMeta
+
+	def __init__(self, config):
+		if config['debug']:
+			print("Initialising Magic Mirro platorm")
+		self.shutdown = False
+		super(MagicmirrorPlatform, self).__init__(config, 'magicmirror')
+		self.req_record = False
+		# get config
+		self.HOST_NAME = self._pconfig['hostname']
+		self.PORT_NUMBER = self._pconfig['port']
+		self.MM_HOST = self._pconfig['mm_hostname']
+		self.MM_PORT = self._pconfig['mm_port']
+		self.HB_TIMER = self._pconfig['hb_timer']
+		self.httpd = ""
+		self.serverthread = ""
+
+	def setup(self):
+		if self._config['debug']:
+			print("Setting up Magic Mirror platform")
+		# Setup http server
+		self.httpd = CallbackHTTPServer((self.HOST_NAME, self.PORT_NUMBER), MMHTTPHandler)
+		self.httpd.set_callback(self.HTTPcallback)
+		print(time.asctime(), "Magic Mirror HTTP Server - %s:%s" % (self.HOST_NAME, self.PORT_NUMBER))
+		self.serverthread = threading.Thread(target=self.httpd.serve_forever)
+		self.serverthread.daemon = True
+
+	def indicate_failure(self):
+		if self._config['debug']:
+			print("Indicating Failure")
+		self.update_mm("failure")
+
+	def indicate_success(self):
+		if self._config['debug']:
+			print("Indicating Success")
+		self.update_mm("success")
+
+	def after_setup(self):
+		# threaded detection of button press
+		if self._config['debug']:
+			print("Starting Magic Mirror platform HTTP Server")
+		self.serverthread.start()
+		# Start HB
+		if self._config['debug']:
+			print("Starting Magic Mirror heartbeat with " + str(self.HB_TIMER) + " second interval")
+		self.mm_heartbeat()
+
+	def indicate_recording(self, state=True):
+		if state:
```

Comment:
Using these forks doesn't look clean to me.
Use something like
```
self.update_mm("recording" if state else "idle")
```

In other places too.